### PR TITLE
Add new saving guide and delete old guide

### PR DESCRIPTION
**Pull request by nkovela1**
This PR adds the new saving guide, titled "Save, serialize, and export", for the TF 2.13 release. It will replace the previous guide titled "Serialization and saving".

This new guide demonstrates the new Keras saving and export APIs, along with some parts carried over from the previous guide.

This PR should be merged at the time of the TF 2.13 release to avoid any confusion.

### CODE REVIEW

File: guides/serialization_and_saving.py [hunk 1]
```diff
@@ -17,18 +16,19 @@
 - A set of weights values (the "state of the model").
 - An optimizer (defined by compiling the model).
 - A set of losses and metrics (defined by compiling the model or calling
-`add_loss()`).
+`add_loss()` or `add_metric()`).
```
Comment (fchollet) [inline guides/serialization_and_saving.py:19]: All references to add_metric should be scrubbed

File: guides/serialization_and_saving.py [hunk 2]
```diff
@@ -17,18 +16,19 @@
...
+marked by the `.keras` extension. This is a zip archive consisting of the
+following:
 
-- Saving everything into a single archive in the TensorFlow SavedModel format
-(or in the older Keras H5 format). This is the standard practice.
-- Saving the architecture / configuration only, typically as a JSON file.
-- Saving the weights values only. This is generally used when training the model.
+- a JSON-based configuration file (config.json): Records of model, layer, and
+other trackables' configuration.
+- NPZ-based state files, found in respective directories, such as
```
Comment (fchollet) [inline guides/serialization_and_saving.py:27]: It's actually h5 based and found in a single file -- `model.weights.h5`

File: guides/serialization_and_saving.py [hunk 3]
```diff
@@ -17,18 +16,19 @@
...
+`model/states.npz`, `model/dense_layer/states.npz`, etc.
+- a Metadata file
```
Comment (fchollet) [inline guides/serialization_and_saving.py:29]: In JSON, storing things such as your Keras version

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: guides/serialization_and_saving.py (hunk 3 above)
Comment (nkovela1) [reply guides/serialization_and_saving.py:29]: Updated, thanks!

File: guides/serialization_and_saving.py (hunk 2 above)
Comment (nkovela1) [reply guides/serialization_and_saving.py:27]: Updated

File: guides/serialization_and_saving.py (hunk 1 above)
Comment (nkovela1) [reply guides/serialization_and_saving.py:19]: Removed those references, thanks!

**Review by fchollet — APPROVED** [review]
Thanks! I'll do a round of edits offline.

**Pull request merged**